### DISCUSSION TITLE
rM version dependency?

### DISCUSSION THREAD

**Discussion by RedTartan04** [slotThe/RemarkableLamyEraser#5] [Q&A]
Hi, a conceptual question:
how (if any) is the relation of this code's version to the rM's software version?
I'm thinking of getting a pen with button, but I have an rm2 with v2.10 and another with the latest stable version (currently 3.7). Can I apply this project's latest version to both devices?
cheers.

**Comment by slotThe** [comment]
> Hi, a conceptual question: how (if any) is the relation of this code's version to the rM's software version?

Oh, very much! This works by simulating touch presses on the remarkable itself, and the configuration of the toolbar can of course change around freely between updates.

> I'm thinking of getting a pen with button, but I have an rm2 with v2.10 and another with the latest stable version (currently 3.7). Can I apply this project's latest version to both devices? cheers.

Only versions with the same toolbar as 3.7 are supported. I guess this should be 3.x at least, possibly even higher.

**Reply by RedTartan04** [reply]
I see, thanks. The touch presses are simluated outside of the xochitl software, I guess, so the only dependency is where on the screen the buttons appear? So for changed toolbars only the coordinates in effect_data.h need to be changed? Maybe I could do that for the rM 2.10.

**Reply by slotThe** [reply]
> The touch presses are simluated outside of the xochitl software, I guess, so the only dependency is where on the screen the buttons appear?

Indeed, yes.

> So for changed toolbars only the coordinates in effect_data.h need to be changed? Maybe I could do that for the rM 2.10.

Sure, sounds great! You can probably rip the [old list](https://github.com/isaacwisdom/RemarkableLamyEraser/blob/v2-3.0.0-fix/headers/effects_data.h) (which I ungracefully just plastered over)  from Isaac's repo and, depending on the [version](https://github.com/slotThe/RemarkableLamyEraser/blob/main/sources/main.c#L29) use that instead

(Really, that whole system is in need of a rewrite, but for now it's good enough, I suppose)

**Comment by RedTartan04** [comment]
I hadn't seen that the v2 branch of the original repo already supports non-eraser tools. So I installed that first, on my v2.10 rM2 with a Samsung pen. Love the features, feels ok but not really stable, the UI stalled once. I'll put this aside for now and get back to your version if/when I'll try it on the other rM with 3.x. Thanks, again!
